src/org/thoughtcrime/securesms/mms/AudioSlide.java

```diff
@@ -47,6 +47,13 @@ public AudioSlide(Context context, Uri uri, long dataSize, String contentType, b
     super(context,  new UriAttachment(uri, null, contentType, AttachmentDatabase.TRANSFER_PROGRESS_STARTED, dataSize, 0, 0, null, null, voiceNote));
   }
 
+  @Override
+  public boolean equals(Object other) {
```

Comment:
this fixes the issue of two different audio messages being considered equal and hence the seekbar and progress listener being set to an old audio message that is not the one being played